### PR TITLE
adding a new method

### PR DESCRIPTION
**Pull request by vlad-ko**

### CODE REVIEW

**Comment by vlad-ko** [comment]
@codecov-ai-reviewer test

**Comment by vlad-ko** [comment]
@codecov-ai-reviewer review

**Comment by ghost** [comment]
On it! We are reviewing the PR and will provide feedback shortly.

File: resources/js/classes/calculator.js [hunk 1]
```diff
@@ -24,6 +24,16 @@ class Calculator {
...
     }
+
+    absolute(number) {
+      return Math.abs(number);
+    }
+
+    factorial(n) {
+      if (n < 0) return NaN;
+      if (n <= 1) return 1;
+      return n * this.factorial(n - 1);
```
Comment (ghost) [inline resources/js/classes/calculator.js:35]: The `factorial` method uses recursion which can lead to stack overflow for large inputs. Consider using an iterative implementation for better performance and to avoid potential stack overflow errors.
```suggestion
factorial(n) {
  if (n < 0) return NaN;
  if (n <= 1) return 1;
  let result = 1;
  for (let i = 2; i <= n; i++) {
    result *= i;
  }
  return result;
}
```

Comment (ghost) [inline resources/js/classes/calculator.js:35]: Missing JSDoc comments for the new methods. Adding documentation would improve code readability and maintainability.
```suggestion
/**
 * Returns the absolute value of a number
 * @param {number} number - The input number
 * @returns {number} The absolute value
 */
absolute(number) {
  return Math.abs(number);
}

/**
 * Calculates the factorial of a non-negative integer
 * @param {number} n - The non-negative integer
 * @returns {number} The factorial value, or NaN for invalid inputs
 */
factorial(n) {
  if (n < 0) return NaN;
  if (n <= 1) return 1;
  return n * this.factorial(n - 1);
}
```

Comment (ghost) [inline resources/js/classes/calculator.js:35]: Consider adding unit tests for the new methods to ensure they work as expected with various inputs, including edge cases.

**Comment by vlad-ko** [comment]
@sentry review

**Comment by vlad-ko** [comment]
@sentry review
